### DISCUSSION TITLE
How to transition out of templates

### DISCUSSION THREAD

**Discussion by Artemis-Hound** [JanDeDobbeleer/oh-my-posh#5836] [Troubleshoot]
I've upgraded to [v24](https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v24.0.0) and I noticed in the breaking changes that templates are now being phased out. I'm not certain how to convert my configuration to deal with this change, for instance one of the sections of my config:

```json
        {
          "properties": {
            "macos": "\uf179",
            "ubuntu": "\uf31b",
            "windows": "\ue62a"
          },
          "foreground": "p:navy",
          "templates_logic": "first_match",
          "background": "p:white",
          "type": "os",
          "style": "diamond",
          "templates": [
            "{{ if .Segments.Session.SSHSession}} {{.Icon}} {{ end }}",
            "{{ if matchP .Var.LinuxOS .OS }}{{.Icon}}{{ end }}",
            "{{.Icon}} "
          ]
        },
```

These templates were here to help me deal with minor spacing issues. You'll notice the template logic is set to first match, and the first template in the list has space on either side of the icon, then no space in the next one, then space only on the right side.

If I read the changelog right, and everything now needs to be its own segment, does this mean the templated section now has to become 3 sections?

**Comment by lewis-yeung** [comment]
> If I read the changelog right, and everything now needs to be its own segment, does this mean the templated section now has to become 3 sections?

@Adam-Kay No. That means you have to use a one-line-only template because the `templates_logic` no longer works since v24. :(

**Comment by JanDeDobbeleer** [comment]
Templates still works, unless I messed up?

**Reply by lewis-yeung** [reply]
The `templates` still works while `templates_logic` not.

**Reply by JanDeDobbeleer** [reply]
check, I'll fix that today. Hopefully that's about the last of things I missed. Never do long lived branches kids, bad idea.

**Reply by Artemis-Hound** [reply]
Seems I misread the changelogs, glad to hear that the template logic might be fixed soon!
